NitroxClient/GameLogic/EscapePodManager.cs

```diff
@@ -28,27 +28,37 @@ public EscapePodManager(IPacketSender packetSender, IMultiplayerSession multipla
             this.multiplayerSession = multiplayerSession;
         }
 
-        public void AssignPlayerToEscapePod(EscapePodModel escapePod)
+        public void AssignPlayerToEscapePod(Optional<EscapePodModel> escapePod)
```

Comment:
Why do we want this to be optional?  There should always be an escape pod for the player.

Reply:
should always be one however there are instances where a packet may be partially sent or drserialized when we get it(throwing that null exception) did check if it made a difference and the answer... the player was able to log into the server fine with the npe not being thrown hell even werewolfs could connect

Reply:
I don't think this is a transmission problem as there are parity checks and the packet should always come in with a good state.  Most likely we have a bug that is causing the player's assigned id to not be sent or incorrectly loading the data.  It could also be a silent failure in deserialization. Irrespective of the cause, we should identify the root problem and fix it; maybe if someone is able to consistently replicate it then identify what piece is missing.

Reply:
And even if we were to take "advantage" of protobufs partial sending / merge / "unpack from stream", the packet should still only be processed when it's complete.

Indeed, this sounds like a mistake elsewhere.

Reply:
Back when testing this... I was able to consistently replicate the issue however I don't remember if It was a protobuf serialization issue(likely) or an issue with other information however as of making this PR it would appear to have been fixed, Having it optional is just for IsEmpty check to use Log.Error